Singularity/Singularity/Sound/SoundManager.cs

```diff
@@ -0,0 +1,482 @@
+                mEffectInstances[mEffectInstanceId].Play();
+                mInstanceMap.Add(mAllInstanceId, new Tuple<SoundClass, int>(soundClass, mEffectInstanceId));
+                mEffectInstanceId++;
+                return mAllInstanceId++;
+            }
+            else if (soundClass == SoundClass.Ui)
+            {
+                SoundEffectInstance effectInstance = mUiSounds[name].CreateInstance();
+                effectInstance.Volume = volume;
+                effectInstance.Pitch = pitch;
+                effectInstance.IsLooped = loop;
+                if (!isGlobal)
+                {
+                    AudioEmitter emitter = new AudioEmitter { Position = new Vector3(x, y, mSoundPlaneDepth) };
+                    effectInstance.Apply3D(mListener, emitter);
+                }
+                mUiInstances.Add(mEffectInstanceId, effectInstance);
+                mUiInstances[mEffectInstanceId].Play();
+                mInstanceMap.Add(mAllInstanceId, new Tuple<SoundClass, int>(soundClass, mUiInstanceId));
+                mUiInstanceId++;
+                return mAllInstanceId++;
+            }
+            else
+            {
+                return -1;
+            }
+        }
+
+        /// <summary>
+        /// Sets the volume of the sound effect instance specified by the given id.
+        /// </summary>
+        /// <param name="id">The global sound instance id.</param>
+        /// <param name="volume">The desired volume.</param>
+        public void SetSoundVolume(int id, float volume)
+        {
+            if (mInstanceMap.ContainsKey(id))
+            {
+                SoundClass soundClass = mInstanceMap[id].Item1;
+                int instanceId = mInstanceMap[id].Item2;
+                if (soundClass == SoundClass.Effect)
+                {
+                    mEffectInstances[instanceId].Volume = volume;
+                }
+                else if (soundClass == SoundClass.Ui)
+                {
+                    mUiInstances[instanceId].Volume = volume;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Setting the position of the Sound Listener used for 3D sound. Uses the given x and y coordinates and the mSoundPlanDepth for the z coordinate.
+        /// </summary>
+        /// <param name="x">New x coordinate of the listener.</param>
+        /// <param name="y">New y coordinate of the listener.</param>
+        public void SetListenerPosition(float x, float y)
+        {
+            mListener.Position = new Vector3(x, y, mSoundPlaneDepth);
+        }
+
+        /// <summary>
+        /// Change the position of a 3D sound effect instance.
+        /// </summary>
+        /// <param name="id">The gloabl id of the sound effect instance.</param>
+        /// <param name="x">The new x coordinate.</param>
+        /// <param name="y">The new y coordinate.</param>
+		public void SetSoundPosition(int id, float x, float y)
```

Comment:
The indentation seems off here ... at least in the webview, can you look into that?

Reply:
webview issue i guess